components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/internal/OpenIDConnectServiceComponent.java

```diff
@@ -94,4 +96,24 @@ protected void unsetIdentityCoreInitializedEventService(IdentityCoreInitializedE
         /* reference IdentityCoreInitializedEvent service to guarantee that this component will wait until identity core
          is started */
     }
+    @Reference(
+            name = "ClaimProvider",
+            service = ClaimProvider.class,
+            cardinality = ReferenceCardinality.OPTIONAL,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetClaimProvider"
+    )
+    protected void setClaimProvider(ClaimProvider claimProvider) {
+        if (log.isDebugEnabled()) {
+            log.debug("Setting ClaimProvider Service");
+        }
+        OpenIDConnectServiceComponentHolder.setClaimProvider(claimProvider);
```

Comment:
Shouldn't this be OpenIDConnectServiceComponentHolder.getClaimProviders().add(claimProvider)